probert/filesystem.py

```diff
@@ -32,6 +32,9 @@ def probe(context=None):
         context = pyudev.Context()
 
     for device in context.list_devices(subsystem='block'):
+        if "MAJOR" not in device:
+            # Shouldn't happen but apparently does! (LP: #1868109)
+            continue
         # Ignore block major=1 (ramdisk) and major=7 (loopback)
         # these won't ever be used in recreating storage on target systems.
         if device['MAJOR'] not in ["1", "7"]:
```

Comment:
I wonder if we should have filtering helper which takes a context and yields devices if they pass these two if checks;  then both files can call the same helper to enumerate the devices they need to operate on?

Reply:
> That feels like a bug. I'm not a fan of this "partially hiding stuff" in the kernel. Why isn't the control node a char device since you can't actually open or interact with it as a block device?

Agreed but I'm even less of a fan of block probe failures :)

Linux block devices, not even once.

> Previous discuss a while back on the curtin side for more background.
> 
> https://bugs.launchpad.net/curtin/+bug/1830913

I *thought* I'd seen some discussion of this sort of thing before.